R/parse.R

```diff
@@ -0,0 +1,109 @@
+
+#' Parse a vector of free text containing drug information
+#'
+#' @description This function provides a dataframe of parsed out strings from a
+#'   free text field given a file path specified by the user.
+#'
+#' @param drug_vec A vector containing the free text to be parsed
+#'
+#' @return A n x 1 dataframe with class \code{data.frame}, \code{tbl_df}, \code{tbl}.
+#'   The dataframe has 1 variable, "drug"
+#' @importFrom utils data
+#' @importFrom stats na.omit
+#' @importFrom magrittr %>%
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' parse(drug_df)
+#' }
+
+parse <- function(drug_vec){
+
+  #binding vars to function
+  word <- drug <- for_token <- stop_words <- NULL
+
+  # drug_data <- df
+
+  ###ID drug col----------------
+
+  # cols <- colnames(drug_data)
+
+  #gather all cols separated by a new line for readline
+  # colOptions <- sapply(1:length(cols), function(x){
+  #   paste0(x, ". ", cols[x])
+  # })
+
+  #function to subset data
+  # get_col <- function(){
+  #   prompt <- stringr::str_c(c("Which column contains drug names?:",colOptions),
+  #                            collapse = " \n")
+  #   colNum <- as.numeric(readline(prompt))
+  #   drug_data[ ,colNum] #returns tibble of n X 1
+  # }
+
+  ####Parse out drug names-------------------------------
+  # data("stop_words", package = "tidytext")
+  drug_stop_words <- c("a", "few", "mg", "pills", "pill","days", "off", "bunch", "street",
+                       "tab", "tabs", "detox", "rx", "not", "unsure", "unknown",
+                       "clinic", "bottle", "unknkwn", "type", "patch", "pm", "which",
+                       "injection", "er", "medication", "mgs", "illicit", "iv", "left",
+                       "patches", "visit", "hcl", "plus", "hd", "bit", "cit",
+                       "sulf", "tart", "c-ject", "es", "hp", "syringe", "contin",
+                       "intensol", "ject", "calcium")
+
+  ##INTERACTIVE - must respond!
+  # drugs <- get_col()
+  #convert vector to df
+  drugs <- as.data.frame(drug_vec)
+  drugCol <- as.name(names(drugs))
```

Comment:
unnecessary to create this variable because you are piping the df into the function on line 67